### ISSUE TITLE
Runtime error after switching between dev-server and watch mode when using filesystem cache

### ISSUE THREAD

**Issue by Raykeen** [webpack-contrib/mini-css-extract-plugin#1028]
### Bug report
Using config with filesystem cache as mini-css-extract-plugin after switching from building project with dev-server to build with --watch I get runtime error `Uncaught TypeError: Cannot read properties of undefined (reading 'dispose') ...` .

### Actual Behavior

After build completes I open builded page and get runtime error:
`Uncaught TypeError: Cannot read properties of undefined (reading 'dispose') ...`
The code of main js entrypoint doesn't run.

### Expected Behavior
No error and main js entrypoint runs.

### How Do We Reproduce?

https://github.com/Raykeen/webpack-dev-watch-switch-hmr-issue

**Comment by alexander-akait** [comment]
Yeah, I will add protection to this, but you need to fix your configuration for better caching, because `dev` and `watch` are different, you need to say webpack invalidate cache:
```
cache: {
        type: "filesystem",
        version: JSON.stringify({
            serve: process.env.WEBPACK_SERVE
        }),
        buildDependencies: {
            config: [__filename],
        }
    },
```

i.e. you invalidate cache between `serve` and `watch`, because you when you run `serve` you have `hot` content

**Comment by alexander-akait** [comment]
Yeah, we will consider it
